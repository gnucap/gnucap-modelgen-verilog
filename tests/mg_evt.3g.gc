
attach ./modelgen_1.so
attach ../vams/vsine.so
attach ./d_reject.so


verilog

`modelgen
module test_timer3g(p, n);
	electrical p, n;
	inout p, n;
	(*desc=""*) real state;
	analog begin
		@(timer(0, 1)) begin
			if(state == 0)	begin
				if(0) $trace("time on %g\n", $abstime);
				state = 1.;
			end else begin
				if(0) $trace("time off %g\n", $abstime);
				state = 0.;
			end
      end
		$write("time0 %g %g\n", $abstime, state);
		V(p, n) <+ state;
	end
endmodule

list

!make test_timer3g.so > /dev/null
attach ./test_timer3g.so

test_timer3g dut0(1,0);
// reject #(.time(.9)) r();

list

print dc v(nodes) state(dut0) iter(0)
print tran v(1) state(dut0) iter(0) control(0)
tran 3 basic
dc
tran 3 basic

end
