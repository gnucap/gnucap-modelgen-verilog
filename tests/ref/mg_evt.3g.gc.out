test_timer3g #() dut0 (.p(1),.n(0));
// reject #(.time(.9)) r(); 
#Time       v(1)       state(dut0) iter(0)    control(0)
time0 0 0
 0.         1.         1.         3.         1.        
time0 0.03 1
 0.03       1.         1.         3.         9.        
time0 1 1
 1.         0.         0.         3.         2.        
time0 2 0
 2.         1.         1.         3.         2.        
time0 3 1
 3.         0.         0.         3.         1.        
#           v(1)       state(dut0) iter(0)   
time0 0 0
 0.         1.         1.         3.        
#Time       v(1)       state(dut0) iter(0)    control(0)
time0 0 1
 0.         0.         0.         3.         1.        
time0 0.03 0
 0.03       0.         0.         3.         9.        
time0 1 0
 1.         1.         1.         3.         2.        
time0 2 1
 2.         0.         0.         3.         2.        
time0 3 0
 3.         1.         1.         3.         1.        
